shell/platform/linux/public/flutter_linux/fl_texture_registrar.h

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_TEXTURE_REGISTRAR_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_TEXTURE_REGISTRAR_H_
+
+#if !defined(__FLUTTER_LINUX_INSIDE__) && !defined(FLUTTER_LINUX_COMPILATION)
+#error "Only <flutter_linux/flutter_linux.h> can be included directly."
+#endif
+
+#include <glib-object.h>
+#include <stdint.h>
+
+G_BEGIN_DECLS
+
+/**
+ * An image buffer object.
+ */
+typedef struct {
+  const uint8_t* buffer;
+  size_t width;
+  size_t height;
+} FlPixelBuffer;
+
+/**
+ * The pixel buffer copy callback provided to the Flutter engine to copy the
+ * texture.
+ */
+typedef const FlPixelBuffer* (*FlTextureCallback)(size_t width,
```

Comment:
You could consider removing `FlPixelBuffer` and do something like:

```
typedef gboolean (*FlTextureCallback)(size_t* width, size_t* height, uint8_t* buffer)
```

Where `width` and `height` are inout parameters, and `buffer` is an out parameter. I say this because its a little unusual to use structs like this is GObject code. If the callback can fail it would also be worth having a `GError**` parameter to allow the callback implementation to report an error that occurred.